crates/rolldown_testing/src/integration_test.rs

```diff
@@ -142,14 +143,20 @@ impl IntegrationTest {
         let cwd = bundler.options().cwd.clone();
         let bundler = Arc::new(Mutex::new(bundler));
 
+        let has_full_reload_update = Arc::new(AtomicBool::new(false));
         let hmr_update_infos = Arc::new(std::sync::Mutex::new(vec![]));
         let build_outputs = Arc::new(std::sync::Mutex::new(vec![]));
         let dev_engine = DevEngine::with_bundler(
           Arc::clone(&bundler),
           DevOptions {
             on_hmr_updates: {
               let hmr_update_infos = Arc::clone(&hmr_update_infos);
+              let has_full_reload_update = Arc::clone(&has_full_reload_update);
```

Comment:
I switched to use atomic bool instead of channel for having intuitive control flow.